src/kinetics/BulkKinetics.cpp

```diff
@@ -120,6 +120,28 @@ bool BulkKinetics::addReaction(shared_ptr<Reaction> r)
     } else {
         m_irrev.push_back(nReactions()-1);
     }
+
+    if (r->newFramework()) {
+        shared_ptr<ReactionRateBase> rate;
+        rate = std::dynamic_pointer_cast<Reaction2>(r)->rate();
```

Comment:
The `newFramework` method isn't really necessary. You can just check the result of the `dynamic_pointer_cast` and if it's 0, then `r` isn't holding a `Reaction2`.

Reply:
:tada: nice!